### DISCUSSION TITLE
Any way to forward an auth session from OAuth system to another system (with API key)?

### DISCUSSION THREAD

**Discussion by cjhines** [ory/oathkeeper#1006] [Q&A] [answered]
**Description**

This is a long shot, but I'm in dire need of advice. If you know of a more appropriate subreddit (or any other community for that matter) for this type of question please share!

I'm working with a legacy OAuth system using email + password grant type (deprecated, I know: it's out of my control).

I have to somehow connect this authentication service with a database backend for which I have an API key for.

---

**Ideal flow**

[Diagram here](https://i.imgur.com/RRmSgui.png)

1. User logs in with OAuth service with email + password, gets access token
2. User sends backend request to some middleware, which does introspection / validates access token
3. Middleware appends API key to request header to retrieve backend data

---

**Things I've considered**

1. [Ory Oathkeeper](https://www.ory.sh/docs/oathkeeper/pipeline/authn#oauth2_introspection)
This seems to expect arguments which would be used for regular OAuth2 redirect flow, such as `pre_authorization`. Also **the OAuth service's introspection endpoint requires client ID and client secret** [as documented here](https://documentation.pingidentity.com/pingfederate/pf83/index.shtml#adminGuide/pf_c_introspectionParameters.html).

2. Traefik instance + [auth forwarding](https://hub.docker.com/r/thomseddon/traefik-forward-auth)
This seems to expect authentication to be done in the middleware rather than passing and introspecting an access token.

**Comment by aeneasr** [comment]
You can probably use the regular bearer token authenticator for this! It doesn’t care how the token was issued, only checks if it is valid!

**Comment by cjhines** [comment]
@aeneasr I don't believe the bearer token authenticator allows for body fields (such as our client_id and client_secret) that are required for the introspection request.

Please correct me if I'm wrong (I hope I am!)

**Comment by cjhines** [comment, marked as answer]
@aeneasr I was able to accomplish this with the feature included in [this MR](https://github.com/ory/oathkeeper/pull/949).
